WPLG-10

Lukyix opened the pull request.

Lukyix committed:
Add basic files for cookiesWall

Lukyix committed:
Add cookiewall styles import to main stylesheet

Imports the cookiewall SCSS module into the main style.scss file to include cookie wall component styles.

Lukyix committed:
Add CookieWall component

Introduces the CookieWall component with basic structure, styling, and buttons for accepting or rejecting cookies. This component displays information about cookie usage and user options.

Lukyix committed:
Add styles for cookie wall component

Introduces SCSS styling for the cookie wall, including layout, responsive design, and visual elements for text, list, and buttons.

Lukyix committed:
Add CookieWall component to App layout

Integrated the CookieWall component into the main App layout to display cookie consent information. This ensures users are informed about cookie usage across the application.

Lukyix committed:
Merge branch 'dev' into WPLG-10

Lukyix committed:
Refactor CookieWall styles and update imports

Updated CookieWall SCSS to use variables and improved BEM structure. Removed unnecessary class from the CookieWall list in the component. Added CookieWall styles import to the main style.scss file for consistent styling.

Lukyix committed:
Refactor CookieWall component and update styles

Renamed BEM block to ms-cookie-wall and improved structure in both TypeScript and SCSS. Added props for custom className, updated markup for better semantic structure, and enhanced button styling. SCSS now uses more specific selectors and improved layout handling for responsiveness.

Lukyix committed:
Update padding on PC and small desktop

Lukyix committed:
Add cookie consent hook and integrate with CookieWall

Introduces useCookiesConsent hook to manage user cookie consent state using localStorage. Updates CookieWall component to use this hook, conditionally render based on consent, and handle accept/reject actions.

SamuelSlavik commented on src/components/cookieWall/CookieWall.tsx:
```diff
@@ -0,0 +1,57 @@
+import { Button } from '@musica-sacra/forms';
+import '@musica-sacra/forms/dist/components/button/button.scss';
+import { useBem } from '@musica-sacra/hooks';
+import './cookiewall.scss';
```
Just a reminder that in plugin this should not be here

SamuelSlavik commented on src/components/cookieWall/useCookieConsent.tsx:
```diff
@@ -0,0 +1,34 @@
+    }, []);
+
+    const reject = useCallback(() => {
+        localStorage.setItem(STORAGE_KEY, 'rejected');
```
If rejected, we cannot save the information to local storage, but to the session storage instead

SamuelSlavik commented on src/components/cookieWall/useCookieConsent.tsx:
```diff
@@ -0,0 +1,34 @@
+        }
+    }, []);
+
+    const accept = useCallback(() => {
```
Maybe slightly better naming of the functions will be good, "accept" is too general, maybe acceptCookies and rejectCookies or even something more specific.

SamuelSlavik commented on src/components/cookieWall/useCookieConsent.tsx:
```diff
@@ -0,0 +1,34 @@
+        isRejected: choice === 'rejected',
+        accept,
+        reject,
+    };
```
We probably do not need to return choice.

And isAccepted/isRejected should be functions(one function isAccepted) that will return true or false but not based on the "choice". The function should instead look into the localStorage/sessionStorage and check if there is the consent accepted or rejected

SamuelSlavik commented on src/components/cookieWall/CookieWall.tsx:
```diff
@@ -0,0 +1,57 @@
+
+    if (!choice) {
+        return null;
+    }
```
After editing the hook, this conditional rendering should be fixed based, cannot be based on the "choice"

Lukyix committed:
Refactor cookie consent hook and storage logic

Renamed consent functions for clarity and added a clearCookiesConsent method. Changed rejected consent to use sessionStorage instead of localStorage. Updated isAccepted and isRejected to be functions checking storage directly, improving consistency and flexibility.

Lukyix committed:
Refactor CookieWall consent logic and handlers

Updated CookieWall to use new consent handler names and acceptance/rejection checks from useCookiesConsent. Removed unused SCSS import and improved conditional rendering for consent state.

SamuelSlavik reviewed (commented):
Review

SamuelSlavik commented on src/components/cookieWall/useCookieConsent.tsx:
```diff
@@ -0,0 +1,48 @@
+  }, []);
+
+  const acceptCookiesConsent = useCallback(() => {
+    localStorage.setItem(STORAGE_KEY, 'accepted');
```
'accepted' and 'rejected' are string constants, we can use macro variable to store the value and then use the macro, just like the STORAGE_KEY

const CONSENT_ACCEPTED = 'accepted'

SamuelSlavik commented on src/components/cookieWall/useCookieConsent.tsx:
```diff
@@ -0,0 +1,48 @@
+    setChoice(null);
+  }, []);
+
+  const isAccepted = useCallback(() => {
```
Still even more precise name should be better, e.g. isCookieConsentAccepted / ....

SamuelSlavik commented on src/components/cookieWall/useCookieConsent.tsx:
```diff
@@ -0,0 +1,48 @@
+    return localStorage.getItem(STORAGE_KEY) === 'accepted';
+  }, []);
+
+  const isRejected = useCallback(() => {
```
If I want to check if the cookies are rejected, I can call the isAccepted function and look for falsy value. This should be names isSet, or better isCookieConsentSet. 

It should check not only for sessionStorage, but also for localStorage and return true if it find anything anywhere

SamuelSlavik commented on src/components/cookieWall/useCookieConsent.tsx:
```diff
@@ -0,0 +1,48 @@
+  }, []);
+
+  const clearCookiesConsent = useCallback(() => {
+    localStorage.removeItem(STORAGE_KEY);
```
This is unecesery function. Rather the reject and accept functions should check the opposite storage and try to clear it.

So, rejectCookiesConsent should check if in the localStorage the cookies are accepted, if yes, it should clear it. The same for the acceptFunction, it should check if in the sessionStorage is something rejected, if yes, it should clear it. So we dont have both rejected and accepted set to true in the local and session storages

SamuelSlavik commented on src/components/cookieWall/useCookieConsent.tsx:
```diff
@@ -0,0 +1,48 @@
+export type CookiesChoice = 'accepted' | 'rejected' | null;
+
+export function useCookiesConsent() {
+  const [choice, setChoice] = useState<CookiesChoice>(null);
```
Completely remove the choice/setChoice and all its instances, it is unnecesery

SamuelSlavik commented on src/components/cookieWall/CookieWall.tsx:
```diff
@@ -0,0 +1,59 @@
+export function CookieWall({ className = '' }: CookieWallProps) {
+    const { acceptCookiesConsent, rejectCookiesConsent, isAccepted ,isRejected} = useCookiesConsent();
+
+  if (isAccepted() || isRejected()) {
```
After changes to the hook, this can be obtained by the one function isSet()

Lukyix committed:
Refactor cookie consent state handling in CookieWall

Replaces separate isAccepted and isRejected checks with a single isCookiesConsentSet property from useCookiesConsent for cleaner logic in the CookieWall component.

Lukyix committed:
Refactor cookie consent hook for improved state handling

Reworks useCookiesConsent to track consent state with a boolean, synchronize state changes via the storage event, and simplify acceptance/rejection logic. Removes unused choice state and isAccepted/isRejected methods, consolidating consent status into isCookiesConsentSet.

Lukyix committed:
Refactor cookie consent logic and simplify state handling

Replaces useState and useEffect-based consent state tracking with direct localStorage/sessionStorage checks. Updates CookieWall to use the new isCookiesConsentSet function, improving reliability and simplifying the code.
